### PR TITLE
test: add tests for handlers

### PR DESCRIPTION
**Pull request by bradgarropy**
## Description

Add tests for event handlers.

## Related Issue

N/A

## Motivation and Context

Increases test coverage!

## How Has This Been Tested?

Tested with the tests I just wrote!

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: tests/contexts/product.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                specialPrice: 14.99,
+                tierPricing: [],
+            },
+            productId: 111111,
+            productType: "normal",
+            sku: "aaaaaa",
+            specialFromDate: "01/01/2021",
+            specialToDate: "01/10/2021",
+            topLevelSku: "bbbbbb",
+            updatedAt: "01/01/2021",
```
Comment (evargast) [inline tests/contexts/product.test.ts:31]: This seems too static and fragile, I'm not too comfortable with these tests 😓

Comment (evargast) [reply tests/contexts/product.test.ts:31]: Could we make them something like:

```javascript 
expect(ctx).toEqual({
    data: { ...mockProductData }
    schema: PRODUCT_SCHEMA_URL
});
```

Or using `.toMatchSchema` and having the JSON schemas listed somewhere

I think having the hard-coded JSONs in every test is not very robust / dynamic.

Comment (evargast) [reply tests/contexts/product.test.ts:31]: Same comment for all tests, they all seem to be hardcoded json comparisons - lets maybe chat / brainstorm on other testing strategies through slack or a bjns call ?

**Review by evargast — CHANGES_REQUESTED** [review]

File: tests/contexts/product.test.ts (hunk 1 above)
Comment (bradgarropy) [reply tests/contexts/product.test.ts:31]: Happy to change this! I was just trying to move fast and get _some_ kind of validation in there.
For starters I'll move them out to `mocks`, does that work?

Comment (evargast) [reply tests/contexts/product.test.ts:31]: Sure! And lets make sure we use the PRODUCT_SCHEMA_URL instead of hardcoding it 🥳

Comment (bradgarropy) [reply tests/contexts/product.test.ts]: Done! Thanks for the candid feedback 👍🏼

**Review by abrhim — APPROVED** [review]

**Review by evargast — APPROVED** [review]
beautiful, thank you!

**Pull request merged**